packages/shared/src/react/types.ts

```diff
@@ -44,6 +44,9 @@ export type PaginatedHookConfig<T> = T & {
    * Return the previous key's data until the new data has been loaded
    */
   keepPreviousData?: boolean;
+
+  __unstable__dependencyRevalidation?: boolean;
+  __unstable__defaultRevalidateOnEvents?: string[];
```

Comment:
These will be exposed to public API. It is the only think that is exposed to public API